docs/src/components/showcase/FolderTreeView.tsx

```diff
@@ -2,209 +2,275 @@
+  TreeItem2Label,
+  TreeItem2Root,
+} from '@mui/x-tree-view/TreeItem2';
+import { TreeItem2Icon } from '@mui/x-tree-view/TreeItem2Icon';
+import { TreeItem2Provider } from '@mui/x-tree-view/TreeItem2Provider';
+import { TreeViewBaseItem } from '@mui/x-tree-view/models';
 
-const CustomContent = React.forwardRef(function CustomContent(
-  props: TreeItemContentProps & { lastNestedChild?: boolean },
-  ref,
-) {
-  const {
-    lastNestedChild,
-    classes,
-    className,
-    label,
-    nodeId,
-    icon: iconProp,
-    expansionIcon,
-    displayIcon,
-  } = props;
+type ExtendedTreeItemProps = {
+  id: string;
+  label: string;
+  color?: 'primary' | 'default';
+};
 
-  const {
-    disabled,
-    expanded,
-    selected,
-    focused,
-    handleExpansion,
-    handleSelection,
-    preventSelection,
-  } = useTreeItem(nodeId);
+const ITEMS: TreeViewBaseItem<ExtendedTreeItemProps>[] = [
+  {
+    id: '1',
+    label: 'src',
+    color: 'primary',
+    children: [
+      {
+        id: '1.1',
+        label: 'components',
+        children: [
+          { id: '1.1.1', label: 'Button.tsx' },
+          { id: '1.1.2', label: 'Drawer.tsx' },
+          { id: '1.1.3', label: 'Navbar.tsx' },
+          { id: '1.1.4', label: 'TreeView.tsx' },
+        ],
+      },
+      {
+        id: '1.2',
+        label: 'blocks',
+        children: [
+          { id: '1.2.1', label: 'SignupPage.tsx' },
+          {
+            id: '1.2.2',
+            label: 'PricingTable',
+            children: [
+              { id: '1.2.2.1', label: 'PaymentOptions.tsx' },
+              { id: '1.2.2.2', label: 'EarlyBirdDiscount.tsx' },
+            ],
+          },
+        ],
+      },
+    ],
+  },
+];
 
-  const icon = iconProp || expansionIcon || displayIcon;
+function DotIcon() {
+  return (
+    <Box
+      sx={{
+        width: 6,
+        height: 6,
+        borderRadius: '70%',
+        bgcolor: 'warning.main',
+        display: 'inline-block',
+        verticalAlign: 'middle',
+        zIndex: 1,
+        mx: 1,
+      }}
+    />
+  );
+}
 
-  const handleMouseDown = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
-    preventSelection(event);
-  };
+declare module 'react' {
+  interface CSSProperties {
+    '--tree-view-color'?: string;
+    '--tree-view-bg-color'?: string;
+  }
+}
 
-  const handleExpansionClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
-    handleExpansion(event);
-  };
+const StyledTreeItemRoot = styled(TreeItem2Root)(({ theme }) => ({
+  color: theme.palette.mode === 'light' ? theme.palette.grey[800] : theme.palette.grey[400],
+  position: 'relative',
+  [`& .${treeItemClasses.groupTransition}`]: {
+    paddingLeft: theme.spacing(0.5),
+  },
+})) as unknown as typeof TreeItem2Root;
+
+const CustomTreeItemContent = styled(TreeItem2Content)(({ theme }) => ({
+  borderRadius: theme.spacing(0.7),
+  marginBottom: theme.spacing(0.5),
+  marginTop: theme.spacing(0.5),
+  padding: theme.spacing(0.5),
```

Comment:
```suggestion
  borderRadius: theme.spacing(0.5),
```

On this one, it seems like we can go on without these extra spacing values and it'll look fine the same way!